keras_cv/utils/preprocessing.py

```diff
@@ -323,3 +324,31 @@ def check_fill_mode_and_interpolation(fill_mode, interpolation):
             "Unknown `interpolation` {}. Only `nearest` and "
             "`bilinear` are supported.".format(interpolation)
         )
+
+
+def clip_bounding_box(bounding_boxes, height, width, bounding_box_format, image=None):
+    """clips bounding boxes to image boundaries"""
+    bounding_boxes = bounding_box.convert_format(
+        bounding_boxes,
+        source=bounding_box_format,
+        target="xyxy",
+        images=image,
+    )
+    x1, y1, x2, y2 = tf.split(bounding_boxes, 4, axis=1)
+    new_bboxes = tf.stack(
+        [
+            tf.clip_by_value(x1, clip_value_min=0, clip_value_max=width),
```

Comment:
we actually wouldn't require height and width parameters if we converted to rel_xyxy coordinates.  Then we can just clip to 0, 1

Reply:
@LukeWood, couple of questions

- in batched input, should we expect image height and width to be different for each input?
- I would need to convert format because the order would be different here for "yxyx"/ "rel_yxyx" formats. Or I could just add the code for this format without converting format.

Reply:
done